protos/flyteidl/core/workflow.proto

```diff
@@ -127,11 +126,30 @@ message Node {
     }
 }
 
-// Metadata for the entire workflow.
-// To be used in the future.
+// Metadata for the entire workflow. Defines execution behavior that does not change the final outputs of the workflow.
 message WorkflowMetadata {
     // Total wait time a workflow can be delayed by queueing.
     google.protobuf.Duration queuing_budget = 1;
+
+    // Failure Handling Strategy
+    enum FailureHandlingStrategy {
+        // IMMEDIATELY instructs the system to fail as soon as a node fails in the workflow. It'll automatically abort
+        // all currently running nodes and clean up resources before finally marking the workflow executions as failed.
+        IMMEDIATELY = 0;
+
+        // WAIT_FOR_RUNNING_NODES instructs the system not to schedule new nodes to run but to wait for all currently
+        // running nodes to finish executing before cleaning up resources and marking the workflow execution as failed.
+        WAIT_FOR_RUNNING_NODES = 1;
+
+        // WAIT_FOR_EXECUTABLE_BRANCHES instructs the system to make as much progress as it can. The system will not
+        // alter the dependencies of the execution graph so any node that depend on the failed node will not be run.
+        // Other nodes that will be executed to completion before cleaning up resources and marking the workflow
+        // execution as failed.
+        WAIT_FOR_EXECUTABLE_BRANCHES = 2;
+    }
+
+    // Defines how the system should behave when a failure is detected in the workflow execution.
+    FailureHandlingStrategy failure_handling_strategy = 2;
```

Comment:
will this be overridable in the launch-plan?

Reply:
done